src/PuJetIdSF.cc

```diff
@@ -218,23 +151,10 @@ float PuJetIdSF::getSFError(bool isReal, float pt, float eta)
 std::vector<float> PuJetIdSF::getEvtWeight(bigTree &theBigTree, TLorentzVector tau1, TLorentzVector tau2, std::map<int,double> jets_and_smearFactor, bool cleanJets)
 {
   // Weight for each event
-  std::vector<float> eventWeight(15);
-  float P_MC   = 1.;
-  float P_DATA = 1.;
-  float P_DATA_up   = 1.;
-  float P_DATA_down = 1.;
-  float P_DATA_effic_up    = 1.;
-  float P_DATA_effic_down  = 1.;
-  float P_DATA_mistag_up   = 1.;
-  float P_DATA_mistag_down = 1.;
-  float P_DATA_effic_eta_s2p5_up    = 1.;
-  float P_DATA_effic_eta_s2p5_down  = 1.;
-  float P_DATA_effic_eta_l2p5_up    = 1.;
-  float P_DATA_effic_eta_l2p5_down  = 1.;
-  float P_DATA_mistag_eta_s2p5_up   = 1.;
-  float P_DATA_mistag_eta_s2p5_down = 1.;
-  float P_DATA_mistag_eta_l2p5_up   = 1.;
-  float P_DATA_mistag_eta_l2p5_down = 1.;
+  std::vector<float> eventWeight(15); // keeping 15 values so that not to change too much skimmers; can be updated
```

Comment:
I would propose to remove the unneeded 12 weights in this PR.

Reply:
Ok, as stated that'd require updating the skimmers as well.

Reply:
Actually, thinking a bit more, this a change that can wait a bit because it is possibile that we will need to split the uncertainties in eta if they are too large so some of the other elements might be needed.